src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/ProblemService.java

```diff
@@ -3,18 +3,107 @@
 
+    @PersistenceContext
+    private EntityManager entityManager;
+
     @Autowired
     private ProblemRepository problemRepository;
 
-    public List<Problem> getProblems() {
-        return problemRepository.findAll();
+    private Function<ProblemReference, ProblemReferenceResponse> problemReferenceDbToResponse =
+        (ProblemReference pr) -> {
+            ProblemReferenceResponse response = new ProblemReferenceResponse();
+            response.setId(pr.getId());
+            response.setType(pr.getType());
+            response.setTypeId(pr.getTypeId());
+            response.setDescription(pr.getDescription());
+            response.setProblemId(pr.getProblem().getId());
+            return response;
+        };
+
+    public Function<Problem, ProblemResponse> problemDbToResponse = (Problem p) -> {
+        ProblemResponse response = new ProblemResponse();
+        response.setId(p.getId());
+        response.setType(p.getType());
+        response.setSeverity(p.getSeverity());
+        response.setReferences(
+            p.getReferences()
+                .stream()
+                .map(problemReferenceDbToResponse)
+                .collect(Collectors.<ProblemReferenceResponse>toList())
+        );
+        return response;
+    };
+
+    private Function<CreateProblemReference, ProblemReference> problemReferenceCreateToDb =
+        (CreateProblemReference cpr) -> {
+            ProblemReference transformed = new ProblemReference();
+            transformed.setId(UUID.randomUUID().toString());
+            transformed.setType(cpr.getType());
+            transformed.setTypeId(cpr.getId());
+            transformed.setDescription(cpr.getDescription());
+            return transformed;
+        };
+
+    public Function<CreateProblem, Problem> problemCreateToDb = (CreateProblem cp) -> {
+        Problem transformed = new Problem();
+        transformed.setId(cp.getId());
+        transformed.setType(cp.getType());
+        transformed.setSeverity(cp.getSeverity());
+        transformed.setReferences(
+            cp.getReferences()
+                .stream()
+                .map(problemReferenceCreateToDb)
+                .collect(Collectors.<ProblemReference>toList())
+        );
+        transformed.getReferences().forEach(pr -> pr.setProblem(transformed));
+        return transformed;
+    };
+
+    public List<ProblemResponse> getProblems() {
+        final List<Problem> problems = problemRepository.findAll();
+
+        return problems.stream()
+            .map(problemDbToResponse)
+            .collect(Collectors.toList());
     }
+
+    public Problem save(Problem problem) {
+        return problemRepository.save(problem);
+    }
+
+    public void removeIfExist(String id) {
+        if (problemRepository.exists(id)) {
+            problemRepository.delete(id);
+        }
+    }
+
+    public List<ProblemResponse> getProblemsByReferenceTypeId(String referenceTypeId) {
+        List<Problem> problems = entityManager
```

Comment:
This query could probably be put in ProblemRepository, Is there really a need for ProblemQueries class?

Reply:
Agree :) @leszekgu please make the change